### PR TITLE
Unsubscribing a topic is removing all topic subscriptions from Qlobber

### PR DESCRIPTION
**Pull request by behrad**
We should add a test for this inside `aedes-persistence`

### CODE REVIEW

**Comment by mcollina** [comment]
Can you add a unit test for this?

**Comment by behrad** [comment]
Sure, that would be another PR in `aedes-persistence`.

**Comment by behrad** [comment]
[aedes-persistence-redis PR](https://github.com/mcollina/aedes-persistence-redis/pull/31) depends on this being merged.
I will also create a PR for a containing test in `aedes-persistence`

**Comment by mcollina** [comment]
I'll wait for the test, I'm a bit busy today to get into a release train.

**Comment by behrad** [comment]
[createRetainedStreamCombi's test](https://github.com/mcollina/aedes-persistence/pull/11)

**Comment by behrad** [comment]
> I'll wait for the test, I'm a bit busy today to get into a release train.

[test added](https://github.com/mcollina/aedes-persistence/pull/11/commits/233927523d196ed7b37e0a04ca7bbda633923b43)
Is this ready now @mcollina ?

**Comment by behrad** [comment]
Have you had time to check this @mcollina ?

**Comment by mcollina** [comment]
@behrad can you update to the new version of aedes-persistence, it's a major version, so you need to update package.json.

**Comment by behrad** [comment]
done @mcollina

**Comment by mcollina** [comment]
CI is not happy :(.
Also, can you add node 8?

**Comment by behrad** [comment]
this is because there's no default implementation for `createRetainedStream` inside CachedPersistence since it is not based upon MemoryPersistence.

**Comment by behrad** [comment]
cached persistence has no parents actually.

**Comment by mcollina** [comment]
Yes, there is no inheritance chain between aedes-persistence and aedes-cached-persistence.
For test perspective, have a look at https://github.com/mcollina/aedes-cached-persistence/blob/master/test.js#L28-L37

**Comment by behrad** [comment]
Yes, now I've doubt where to put 
```js
XPersistence.prototype.createRetainedStreamCombi = function (patterns) {
  var that = this
  var streams = patterns.map(function (p) {
    return that.createRetainedStream(p)
  })
  return MultiStream(streams)
}
```
Should `XPersistence` be CachedPersistence or MemoryPersistence? And would that help mongo, nedb to be made compatible @mcollina ?

**Comment by mcollina** [comment]
As there is not inheritance between `aedes-persistence` (that's just an interface), add it here.

**Comment by behrad** [comment]
this should now be ready @mcollina

File: test.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ util.inherits(MyPersistence, CachedPersistence)
   'outgoingUpdate', 'outgoingClearMessageId',
   'incomingStorePacket', 'incomingGetPacket',
   'incomingDelPacket', 'putWill', 'delWill',
-  'createRetainedStream', 'outgoingStream',
-  'subscriptionsByClient',
+  'createRetainedStream', 'createRetainedStreamCombi',
```
Comment (mcollina) [inline test.js]: then `createRetainedStreamCombi` is not needed here?

Comment (behrad) [inline test.js]: this is added to makes test pass, there it is!

Comment (mcollina) [inline test.js]: [this](https://github.com/mcollina/aedes-cached-persistence/pull/4/files#diff-168726dbe96b3ce427e7fedce31bb0bcR143) should make it pass.

Comment (behrad) [inline test.js]: I got it right now @mcollina

**Comment by behrad** [comment]
Is this ok to be merged @mcollina ?

**Pull request merged**